src/server/qgsowsserver.cpp

```diff
@@ -64,3 +64,18 @@ void QgsOWSServer::restoreLayerFilters( const QHash<QgsMapLayer*, QString>& filt
     }
   }
 }
+
+QString QgsOWSServer::featureGmlId( const QgsFeature* f, const QgsAttributeList& pkAttributes )
+{
+  if ( !f )
+  {
+    return QString();
+  }
+
+  if ( pkAttributes.size() != 1 )
+  {
+    return QString::number( f->id() );
+  }
+
+  return f->attribute( pkAttributes.at( 0 ) ).toString();
```

Comment:
I think it will be usefull to concatenate all pkAttributes with `'.'` and not only used the first one.

Reply:
I'd also like to use all pkAttributes. Is it safe to use '.' as a delimiter? E.g. if a string attribute contains '.', there might be a problem in WFS transaction when splitting.

Reply:
which character will have your preference: `_`, `.`, `-` ?

Reply:
I don't know. It could also be several characters, it should just be very unlikely they appear together in an attribute value

Reply:
like `_-+-_` ?

Reply:
or `@@` or `||`

Reply:
'@@' looks unlikely enough

Reply:
Yes `@@` is enough